File: work/deprecated/room/converter/WeatherSourceConverter.java
package wangdaye.com.geometricweather.room.converter;

import androidx.room.TypeConverter;

import wangdaye.com.geometricweather.basic.models.options.provider.WeatherSource;

public class WeatherSourceConverter {

    @TypeConverter
    public static WeatherSource convertToEntityProperty(String databaseValue) {
        return WeatherSource.valueOf(databaseValue);
    }

    @TypeConverter
    public static String convertToDatabaseValue(WeatherSource entityProperty) {
        return entityProperty.name();
    }
}

File: app/src/main/java/org/breezyweather/db/converters/WeatherSourceConverter.kt
package org.breezyweather.db.converters

import io.objectbox.converter.PropertyConverter
import org.breezyweather.common.basic.models.options.provider.WeatherSource

class WeatherSourceConverter : PropertyConverter<WeatherSource, String?> {

    override fun convertToEntityProperty(databaseValue: String?) =
        // use get instance method but not getValue method.
        WeatherSource.getInstance(databaseValue)

    override fun convertToDatabaseValue(entityProperty: WeatherSource) =
        entityProperty.id
}